packages/smooth_app/lib/pages/product/common/product_query_page.dart

```diff
@@ -410,34 +408,24 @@ class _ProductQueryPageState extends State<ProductQueryPage>
         ),
       );
 
-  _Action _getRefreshAction(
-    final AppLocalizations appLocalizations,
-  ) =>
-      _Action(
-        text: appLocalizations.label_refresh,
-        iconData: Icons.refresh,
-        onPressed: () async {
-          final bool? success =
-              await _loadAndRefreshDisplay(_model.loadFromTop());
-          if (success == true) {
-            _scrollToTop(instant: true);
-          }
-        },
-      );
-
   void retryConnection() =>
       setState(() => _model = _getModel(widget.productListSupplier));
 
   ProductQueryModel _getModel(final ProductListSupplier supplier) =>
       ProductQueryModel(supplier);
 
   Future<void> refreshList() async {
-    final ProductListSupplier? refreshSupplier =
-        widget.productListSupplier.getRefreshSupplier();
-    setState(
-      // How do we refresh a supplier that has no refresher? With itself.
-      () => _model = _getModel(refreshSupplier ?? widget.productListSupplier),
-    );
+    bool successfullyLoaded = false;
+    try {
+      successfullyLoaded = await _model.loadFromTop();
+    } catch (e) {
+      //Add snackbar to show error?
+    } finally {
+      if (successfullyLoaded) {
+        _scrollToTop(instant: true);
```

Comment:
Side note: as you removed the button, the scroll to top does not make sense anymore as it's a pull-to-refresh gesture that triggers the refresh. We're already on top.

Reply:
I added it because user can still scroll while page is refreshing (though we can restrict scrolling while it's refreshing?)